ghcide/test/exe/HieDbRetry.hs

```diff
@@ -5,25 +5,34 @@ import           Control.Concurrent.Extra     (Var, modifyVar, newVar, readVar,
                                                withVar)
 import           Control.Exception            (ErrorCall (ErrorCall), evaluate,
                                                throwIO, tryJust)
-import           Data.Text                    (Text)
+import           Control.Monad.IO.Class       (MonadIO (liftIO))
 import           Data.Tuple.Extra             (dupe)
 import qualified Database.SQLite.Simple       as SQLite
 import           Development.IDE.Session      (retryOnException,
                                                retryOnSqliteBusy)
-import           Development.IDE.Types.Logger (Logger (Logger), Priority,
-                                               noLogging)
+import qualified Development.IDE.Session      as Session
+import           Development.IDE.Types.Logger (Recorder (Recorder, logger_),
+                                               WithPriority (WithPriority, payload),
+                                               cmapWithPrio)
 import qualified System.Random                as Random
 import           Test.Tasty                   (TestTree, testGroup)
 import           Test.Tasty.HUnit             (assertFailure, testCase, (@?=))
 
-makeLogger :: Var [(Priority, Text)] -> Logger
-makeLogger msgsVar = Logger $ \priority msg -> modifyVar msgsVar (\msgs -> pure ((priority, msg) : msgs, ()))
+data Log
+  = LogSession Session.Log
+  deriving Show
+
+makeLogger :: Var [Log] -> Recorder (WithPriority Log)
```

Comment:
Probably not for this PR to keep it minimal, but there's a nice generic version of this backed by a `TQueue` or something, so you could have
```
mkTQueueLogger :: IO (Recorder msg, TQueue msg)
```
or something.

Reply:
That's more convenient. It never occurred to me that the recorder is kind of like the in pipe so you can create them both at same time.